### PR TITLE
Further docs improvements and full spellcheck

### PR DESCRIPTION
**Pull request by daquinteroflex**
* Emoji drop
* m2r2 conversion.
* Full spellcheck of the repository

### CODE REVIEW

**Comment by daquinteroflex** [comment]
I think ruff screwed up some tests

File: CHANGELOG.md [hunk 1]
```diff
@@ -945,7 +945,7 @@ which fields are to be projected is now determined automatically based on the me
 - PML parameters and padding Grid with pml pixels by [@momchil-flex](https://github.com/momchil-flex) in #64
 - Documentation by [@tylerflex](https://github.com/tylerflex) in #63
 - Gds import from [@tylerflex](https://github.com/tylerflex) in #69
-- Loggin by [@tylerflex](https://github.com/tylerflex) in #70
+- Login, by [@tylerflex](https://github.com/tylerflex) in #70
```
Comment (tylerflex) [inline CHANGELOG.md:948]: I think this is supposed to be "Logging"

**Review by tylerflex — COMMENTED** [review]
Thanks @daquinteroflex !
There were a lot of typos :D. Also will be fun to have all of these emojis.

About the m2r2, do we need to automatically convert the markdown files to readme every time we change the readme? or how will that work?

**Comment by daquinteroflex** [comment]
Whoops so any remaining rst converted files are left as a mistake. `m2r2` should be working as before with `.. mdinclude` which was what I was working on to make compatible with sphinx6 and most relevant features are converted (not all but I can implement as necessary little by little).  

I will remove the generated rst files and fix the error you mention above and rebase then.

**Pull request merged**